component/file_cache/file_cache_test.go

```diff
@@ -1588,24 +1599,27 @@ func (suite *fileCacheTestSuite) TestHardLimitOnSize() {
 	err = os.WriteFile(suite.fake_storage_path+"/"+pathsmall, data, 0777)
 	suite.assert.NoError(err)
 
+	smallHandle, err := suite.fileCache.OpenFile(internal.OpenFileOptions{Name: pathsmall, Flags: os.O_RDONLY, Mode: suite.fileCache.defaultPermission})
+	suite.assert.NoError(err)
 	// try opening small file
-	options := internal.OpenFileOptions{Name: pathsmall, Mode: 0777}
-	f, err := suite.fileCache.OpenFile(options)
+	err = suite.fileCache.downloadFile(smallHandle)
 	suite.assert.NoError(err)
-	suite.assert.False(f.Dirty())
-	err = suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: f})
+	suite.assert.False(smallHandle.Dirty())
+	err = suite.fileCache.CloseFile(internal.CloseFileOptions{Handle: smallHandle})
 	suite.assert.NoError(err)
 
+	bigHandle, err := suite.fileCache.OpenFile(internal.OpenFileOptions{Name: pathbig, Flags: os.O_RDONLY, Mode: suite.fileCache.defaultPermission})
+	suite.assert.NoError(err)
 	// try opening bigger file which shall fail due to hardlimit
-	options = internal.OpenFileOptions{Name: pathbig, Mode: 0777}
-	f, err = suite.fileCache.OpenFile(options)
+	err = suite.fileCache.downloadFile(bigHandle)
```

Comment:
I think we should handle the issue this test brings up without downloading. They want to prevent Opening a file when the download limit is reached. The user should probably see this error on Open otherwise the reads will fail and it won't be completely transparent why. I would make a change in OpenFile to do the check for size there. There you could return the expected error.

Reply:
This will also allow the handle to be nil as originally tested. Great suggestion. I'll get that implemented.